web3/_utils/contracts.py

```diff
@@ -294,120 +222,102 @@ def prepare_transaction(
     fn_abi: ABIFunction,
     data: HexStr,
     normalizers: Sequence[Callable[[TypeStr, Any], Tuple[TypeStr, Any]]] = None,
 ) -> Dict[str, Any]:
-    data = HexBytes(data)
+    data_bytes = HexBytes(data)
     types = get_abi_input_types(fn_abi)
     abi_codec = ABICodec(default_registry)
-    decoded = abi_codec.decode(types, HexBytes(data[4:]))
+    decoded = abi_codec.decode(types, data_bytes[4:])
     if normalizers:
         decoded = map_abi_data(normalizers, types, decoded)
     return named_tree(fn_abi["inputs"], decoded)
 
 
-def get_fallback_function_info(
-    contract_abi: Optional[ABI] = None, fn_abi: Optional[ABIFunction] = None
-) -> Tuple[ABIFunction, HexStr, Tuple[Any, ...]]:
+def get_constructor_function_info(
+    contract_abi: Optional[ABI] = None, fn_abi: Optional[ABIConstructor] = None
+) -> Tuple[ABIConstructor, HexStr, Tuple[Any, ...]]:
     if fn_abi is None:
-        fn_abi = get_fallback_func_abi(contract_abi)
+        fn_abi = get_constructor_function_abi(contract_abi)
     fn_selector = encode_hex(b"")
     fn_arguments: Tuple[Any, ...] = tuple()
     return fn_abi, fn_selector, fn_arguments
 
 
-def get_receive_function_info(
-    contract_abi: Optional[ABI] = None, fn_abi: Optional[ABIFunction] = None
-) -> Tuple[ABIFunction, HexStr, Tuple[Any, ...]]:
+def get_fallback_function_info(
+    contract_abi: Optional[ABI] = None, fn_abi: Optional[ABIFallback] = None
+) -> Tuple[ABIFallback, HexStr, Tuple[Any, ...]]:
     if fn_abi is None:
-        fn_abi = get_receive_func_abi(contract_abi)
+        fn_abi = get_fallback_function_abi(contract_abi)
     fn_selector = encode_hex(b"")
     fn_arguments: Tuple[Any, ...] = tuple()
     return fn_abi, fn_selector, fn_arguments
 
 
-def get_function_info(
-    fn_name: str,
-    abi_codec: ABICodec,
-    contract_abi: Optional[ABI] = None,
-    fn_abi: Optional[ABIFunction] = None,
-    args: Optional[Sequence[Any]] = None,
-    kwargs: Optional[Any] = None,
-) -> Tuple[ABIFunction, HexStr, Tuple[Any, ...]]:
-    if args is None:
-        args = tuple()
-    if kwargs is None:
-        kwargs = {}
-
+def get_receive_function_info(
+    contract_abi: Optional[ABI] = None, fn_abi: Optional[ABIReceive] = None
+) -> Tuple[ABIReceive, HexStr, Tuple[Any, ...]]:
     if fn_abi is None:
-        fn_abi = find_matching_fn_abi(contract_abi, abi_codec, fn_name, args, kwargs)
-
-    fn_selector = encode_hex(function_abi_to_4byte_selector(fn_abi))
-
-    fn_arguments = merge_args_and_kwargs(fn_abi, args, kwargs)
-
-    _, aligned_fn_arguments = get_aligned_abi_inputs(fn_abi, fn_arguments)
-
-    return fn_abi, fn_selector, aligned_fn_arguments
+        fn_abi = get_receive_function_abi(contract_abi)
+    fn_selector = encode_hex(b"")
+    fn_arguments: Tuple[Any, ...] = tuple()
+    return fn_abi, fn_selector, fn_arguments
 
 
-def validate_payable(transaction: TxParams, abi: ABIFunction) -> None:
+def validate_payable(transaction: TxParams, abi_element: ABICallable) -> None:
```

Comment:
nit, distinguish the input name as the type to avoid confusion with `ABIElement` / `abi_element` methods.


```suggestion
def validate_payable(transaction: TxParams, abi_callable: ABICallable) -> None:
```